### PR TITLE
Fix some typo and warning found by intellij idea

### PR DESCRIPTION
**Pull request by remilauzier**

### CODE REVIEW

File: src/common.rs [hunk 1]
```diff
@@ -165,7 +165,7 @@ pub fn solve_quadratic(c0: f64, c1: f64, c2: f64) -> ArrayVec<[f64; 2]> {
 /// represent the lower and upper bounds of the bracket searched for a
 /// solution.
 ///
-/// The ITP method has tuning parameters. This implementation hardwires
+/// The ITP method has tuning parameters. This implementation hardwired
```
Comment (raphlinus) [inline src/common.rs:168]: In this case, "hardwires" is better, I think.

**Review by raphlinus — APPROVED** [review]
Thanks! Most of these (with one small exception) are improvements. I think many of the commas are judgment calls, but are fine.

File: src/arc.rs [hunk 2]
```diff
@@ -136,16 +136,16 @@ impl Shape for Arc {
         iter::once(PathEl::MoveTo(self.center + p0)).chain(self.append_iter(tolerance))
     }
 
-    /// Note: shape isn't closed so area is not well defined.
+    /// Note: shape isn't closed so area is not well-defined.
```
Comment (cmyr) [inline src/arc.rs:139]:
```suggestion
    /// Note: shape isn't closed so area is not well defined.
```

https://english.stackexchange.com/questions/410283/hyphen-well-defined-vs-well-defined

File: src/circle.rs [hunk 3]
```diff
@@ -103,7 +103,7 @@ impl Shape for Circle {
         } else {
             // This is empirically determined to fall within error tolerance.
             let n = (1.1163 * scaled_err).powf(1.0 / 6.0).ceil() as usize;
-            // Note: this isn't minimum error, but it is simple and we can easily
+            // Note: this isn't minimum error, but it is simple, and we can easily
```
Comment (cmyr) [inline src/circle.rs:106]:
```suggestion
            // Note: this isn't minimum error, but it is simple and we can easily
```

this works well enough without the comma splice :)

File: src/cubicbez.rs [hunk 4]
```diff
@@ -235,7 +235,7 @@ impl ParamCurveArclen for CubicBez {
         // A degenerate curve where all points are identical will cause infinite recursion in the rec function (well, until MAX_DEPTH at least) in all branches.
         // This test will in addition be true if the bezier curve is just a simple line (i.e. p0=p1 and p2=p3).
         // The constant 0.5 has not been mathematically proven to be small enough, but from empirical tests
-        // a value of about 0.87 should be enough. Thus 0.5 is a conservative value.
+        // a value of about 0.87 should be enough. Thus, 0.5 is a conservative value.
```
Comment (cmyr) [inline src/cubicbez.rs:238]:
```suggestion
        // a value of about 0.87 should be enough. Thus 0.5 is a conservative value.
```

ditto

File: src/ellipse.rs [hunk 5]
```diff
@@ -13,8 +13,8 @@ use crate::{Affine, Arc, ArcAppendIter, Circle, PathEl, Point, Rect, Shape, Size
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct Ellipse {
     /// All ellipses can be represented as an affine map of the unit circle,
-    /// centred at (0, 0). Therefore we can store the ellipse as an affine map,
-    /// with the implication it be applied to the unit circle to recover the
+    /// centred at (0, 0). Therefore, we can store the ellipse as an affine map,
```
Comment (cmyr) [inline src/ellipse.rs:16]:
```suggestion
    /// centred at (0, 0). Therefore we can store the ellipse as an affine map,
```

File: src/ellipse.rs [hunk 6]
```diff
@@ -13,8 +13,8 @@ use crate::{Affine, Arc, ArcAppendIter, Circle, PathEl, Point, Rect, Shape, Size
...
+    /// with the implication it is applied to the unit circle to recover the
```
Comment (cmyr) [inline src/ellipse.rs:17]:
```suggestion
    /// with the implication that it be applied to the unit circle to recover the
```

File: src/ellipse.rs [hunk 7]
```diff
@@ -117,7 +117,7 @@ impl Ellipse {
 
     /// Returns the two radii of this ellipse.
     ///
-    /// The first number is the horizontal radius and the second is the vertical
+    /// The first number is the horizontal radius, and the second is the vertical
```
Comment (cmyr) [inline src/ellipse.rs:120]:
```suggestion
    /// The first number is the horizontal radius and the second is the vertical
```

The trailing comma is to join the dependent clause "before rotation", but a simple "and" works here

File: src/shape.rs [hunk 8]
```diff
@@ -55,7 +55,7 @@ pub trait Shape: Sized {
     /// Convert to a Bézier path.
     ///
     /// This always allocates. It is appropriate when both the source
-    /// shape and the resulting path are to be retained.
+    /// shape, and the resulting path are to be retained.
```
Comment (cmyr) [inline src/shape.rs:58]:
```suggestion
    /// shape and the resulting path are to be retained.
```

**Review by cmyr — APPROVED** [review]
Thanks, you've caught a lot of typos here.

I'm going to pass on some of the additional commas, generally where the word "and" is already serving to join two clauses. I'll go ahead and change those back myself and then merge.

Thanks again!

File: src/arc.rs [hunk 9]
```diff
@@ -154,7 +154,7 @@ impl Shape for Arc {
         self.path_segments(0.1).perimeter(accuracy)
     }
 
-    /// Note: shape isn't closed so a point's winding number is not well defined.
+    /// Note: shape isn't closed, so a point's winding number is not well-defined.
```
Comment (cmyr) [inline src/arc.rs:157]:
```suggestion
    /// Note: shape isn't closed, so a point's winding number is not well defined.
```

**Pull request merged**